pkg/build/controller/build/build_controller.go

```diff
@@ -1978,7 +2014,15 @@ func (bc *BuildController) readClusterImageConfig() []error {
 		bc.setAdditionalTrustedCAs(additionalCAs)
 	}
 
-	registriesTOML, regErr := bc.createBuildRegistriesConfigData(imageConfig)
+	imageContentSourcePolicy, err := bc.imageContentSourcePolicyLister.Get("cluster")
```

Comment:
There may be many ICSP objects, including multiple ICSP objects specifying the same `mirror.Source` which need to be combined as documented.

Reply:
Interesting ... @adambkaplan 's https://bugzilla.redhat.com/show_bug.cgi?id=1745192#c1 implied there would be a cluster wide object.

We'll need to reconcile that.  But yeah, if there are many vs. a singleton, that will impact the controller's informer flow, with separate event handlers for the ICSP objects, and probably a separate queue as well.

Reply:
I'm exploring the MCO now using `updateRegistriesConfig` as a starting point to fully see how it fetches and munges the imageconfig and icsp's  .... based on what I'm seeing so far, don't think I'll have to divert from the current controller informer flow like I said yesterday.